packages/terra-spacer/src/Spacer.jsx

```diff
@@ -4,108 +4,140 @@ import classNames from 'classnames/bind';
 import 'terra-base/lib/baseStyles';
 import styles from './Spacer.scss';
 
+import { mapShorthandToObject, shorthandValidator } from './_spacerShorthandUtils';
+import { SpacerSizes, SpacerClassMappings } from './_spacerConstants';
+
 const cx = classNames.bind(styles);
-const values = ['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'];
+
+/*
+  NOTE: this is needed in order to ensure that the types `marginTop`, `marginRight`, `paddingTop`, `paddingRight`, etc. are recognized by terra-props-table
+  as defined. Currently, simply putting `Object.values(SpacerSizes)` causes all other `marginX` and `paddingX` props to be recognized as `undefined`
+  in the PropsTable.
+*/
+const arrayOfSpacerSizes = [...Object.values(SpacerSizes)];
 
 const propTypes = {
   /**
    * Child Nodes.
    */
   children: PropTypes.node.isRequired,
+  /**
+   * Sets margin in a syntax similar to the standard CSS spec https://developer.mozilla.org/en-US/docs/Web/CSS/margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  margin: shorthandValidator,
   /**
    * Sets top margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
    */
-  marginTop: PropTypes.oneOf(values),
+  marginTop: PropTypes.oneOf(arrayOfSpacerSizes),
   /**
    * Sets bottom margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
    */
-  marginBottom: PropTypes.oneOf(values),
+  marginBottom: PropTypes.oneOf(arrayOfSpacerSizes),
   /**
    * Sets left margin One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
    */
-  marginLeft: PropTypes.oneOf(values),
+  marginLeft: PropTypes.oneOf(arrayOfSpacerSizes),
   /**
    * Sets right margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
    */
-  marginRight: PropTypes.oneOf(values),
+  marginRight: PropTypes.oneOf(arrayOfSpacerSizes),
+  /**
+   * Sets padding in a syntax similar to the standard CSS spec https://developer.mozilla.org/en-US/docs/Web/CSS/padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
+   */
+  padding: shorthandValidator,
```

Comment:
This should be updated to be of PropTypes.string and we should be checking if the values provided are valid in the `mapShorthandToObject` method.

Reply:
@bjankord and I actually talked about this here: https://github.com/cerner/terra-core/issues/1314#issuecomment-377193957

Reply:
Gottchya, I would note this is a string value in the description to keep things clear